code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -526,15 +526,14 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 	if last := allocChallenges.LatestCompletedChallenge; last != nil {
 		latestCompletedChallTime = last.Created
 	}
-
+	blobAlloc.LatestCompletedChallTime = latestCompletedChallTime
 	challenge.Responded = true
 	cab := &challengeAllocBlobberPassResult{
-		verifyTicketsResult:      result,
-		alloc:                    alloc,
-		allocChallenges:          allocChallenges,
-		challenge:                challenge,
-		blobAlloc:                blobAlloc,
-		latestCompletedChallTime: latestCompletedChallTime,
```

Comment:
We can keep the `latestCompletedChallTime` here to keep the changes minimal, but remember to update the `blobAlloc.LatestCompletedChallenge` when the challenge is passed, check the [code](https://github.com/0chain/0chain/blob/c2c4eeb315963332a7866d17b1fe60524993ce4e/code/go/0chain.net/smartcontract/storagesc/models.go#L163) where the `LatestCompletedChallenge` in `AllocationChallenge` is updated, and see how can we update the blobAlloc.